template/alpha_formula.py

```diff
@@ -31,16 +32,18 @@ def __init__(self, _startdate, data, parameters):
             for instrument in data['currency_price_tw']
         }
 
-    def generate(self, date, data):
+    def generate(self, _date, data):
         """
         Generate portfolio for the given date.
         The simulator guarantees to call this method in date order, and the
         first date will be "startdate" specified in AlphaSettings.
         """
-        portfolio = Portfolio(date)
+        portfolio = Portfolio()
         for instrument, price_data in data['currency_price_tw'].items():
             # maintain mid_last_k_days and sum_last_k_days for optimization.
-            new = (price_data[-1].buy + price_data[-1].sell) / 2
+            new = ((price_data[-1].buy + price_data[-1].sell) / 2
+                   if price_data[-1].buy is not None and
+                   price_data[-1].sell is not None else 0)
```

Comment:
Same as above.

Reply:
In real KDR may not write as this (append 0 when Data is None). However, it is more easier to write, so I think do this for template is enough.

Reply:
Well, if any of buy and sell is None, we WILL NOT default them by 0. This will destroy the alpha just because of lacking data.
I think a better way is not to update `mid_last_k_days` and `sum_last_k_days` at all in the case.